tests/unit/task_processor/test_unit_task_processor_processor.py

```diff
@@ -63,6 +70,83 @@ def test_run_task_runs_task_and_creates_task_run_object_when_success(db):
     assert task.completed
 
 
+def test_run_task_kills_task_after_timeout(
+    db: None,
+    get_task_processor_caplog: "GetTaskProcessorCaplog",
+) -> None:
+    # Given
+    caplog = get_task_processor_caplog(logging.ERROR)
+    task = Task.create(
+        _sleep.task_identifier,
+        scheduled_for=timezone.now(),
+        args=(1,),
+        timeout=timedelta(microseconds=1),
+    )
+    task.save()
+
+    # When
+    task_runs = run_tasks()
+
+    # Then
+    assert len(task_runs) == TaskRun.objects.filter(task=task).count() == 1
+    task_run = task_runs[0]
+    assert task_run.result == TaskResult.FAILURE
+    assert task_run.started_at
+    assert task_run.finished_at is None
+    assert "TimeoutError" in task_run.error_details
+
+    task.refresh_from_db()
+
+    assert task.completed is False
+    assert task.num_failures == 1
+    assert task.is_locked is False
+
+    assert len(caplog.records) == 1
+    assert caplog.records[0].message == (
+        f"Failed to execute task '{task.task_identifier}', with id {task.id}. Exception: TimeoutError()"
+    )
+
+
+def test_run_recurring_task_kills_task_after_timeout(
+    db: None,
+    monkeypatch: MonkeyPatch,
+    get_task_processor_caplog: "GetTaskProcessorCaplog",
+) -> None:
+    # Given
+    caplog = get_task_processor_caplog(logging.ERROR)
+    monkeypatch.setenv("RUN_BY_PROCESSOR", "True")
+
+    @register_recurring_task(
+        run_every=timedelta(seconds=1), timeout=timedelta(microseconds=1)
+    )
+    def _dummy_recurring_task():
+        time.sleep(1)
+
+    task = RecurringTask.objects.get(
+        task_identifier=_dummy_recurring_task.task_identifier
+    )
+    # When
+    task_runs = run_recurring_tasks()
+
+    # Then
+    assert len(task_runs) == RecurringTaskRun.objects.filter(task=task).count() == 1
+    task_run = task_runs[0]
+    assert task_run.result == TaskResult.FAILURE
+    assert task_run.started_at
+    assert task_run.finished_at is None
+    assert "TimeoutError" in task_run.error_details
+
+    task.refresh_from_db()
+
+    assert task.locked_at is None
+    assert task.is_locked is False
```

Comment:
Minor: maybe we should add an assert on `task.locked_at is None`?

Reply:
Yep